src/v0/destinations/salesforce/utils.js

```diff
@@ -28,72 +25,74 @@ const ACCESS_TOKEN_CACHE = new Cache(ACCESS_TOKEN_CACHE_TTL);
 const salesforceResponseHandler = (destResponse, sourceMessage, authKey, authorizationFlow) => {
   const { status, response } = destResponse;
 
-  // if the response from destination is not a success case build an explicit error
-  if (!isHttpStatusSuccess(status) && status >= 400) {
-    const matchErrorCode = (errorCode) =>
-      response && Array.isArray(response) && response.some((resp) => resp?.errorCode === errorCode);
-    if (status === 401 && authKey && matchErrorCode('INVALID_SESSION_ID')) {
-      if (authorizationFlow === OAUTH) {
+  const matchErrorCode = (errorCode) =>
+    response && Array.isArray(response) && response.some((resp) => resp?.errorCode === errorCode);
+
+  const getErrorMessage = () => {
+    if (response && Array.isArray(response) && response[0]?.message?.length > 0) {
```

Comment:
add comment about this condition, explaining what is this checking

Reply:
done